the-collab-lab/tcl-16-smart-shopping-list
# Issue #1 As a component, I want to be able to read from and write to the Firestore database so that users can persist information.

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

- adds firebase & react-firebase-hooks dependencies 
- connects to firestore to allow user interactivity 
- user can make a change in the Firestore database and it shows up in the app
- user can make a change in the app and it shows up in the Firestore database
- users can add and delete items from the shopping list 

## Related Issue

closes #1 

## Acceptance Criteria

AC:

The following have been added as project dependencies: firebase & react-firebase-hooks
Make a change in the Firestore database and it shows up in the app
Make a change in the app and it shows up in the Firestore database

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

N/A - basic react page 

### After

![Screen Shot 2021-01-07 at 8 38 59 PM](https://user-images.githubusercontent.com/46036289/103964030-72b5f080-5128-11eb-917b-85bdfb5c2b73.png)

[![Image from Gyazo](https://i.gyazo.com/52f4f39c13da9e83113b9031840c6492.gif)](https://gyazo.com/52f4f39c13da9e83113b9031840c6492)

[![Image from Gyazo](https://i.gyazo.com/02c49e96577f03aef2c1af2b6da07cb0.gif)](https://gyazo.com/02c49e96577f03aef2c1af2b6da07cb0)

## Testing Steps / QA Criteria

1. pull from our branch
2. npm install to update dependencies 
3. open app and also firestore database 
4. add a new item in app, and check firestore database to ensure it appears 
5. click on item in app to delete the item, and check firestore to ensure it is removed
